src/base/bittorrent/sessionimpl.cpp

```diff
@@ -2807,6 +2820,16 @@ void SessionImpl::findIncompleteFiles(const TorrentInfo &torrentInfo, const Path
     });
 }
 
+void SessionImpl::invoke(std::function<void ()> func)
+{
```

Comment:
I have already approved this PR. But here is a suggestion for a different incantation. Do what you want with it.
I noticed that QMetaObject::invokeMethod() is a templated method. So invoke() could be implemented like this too:
```c++
template<typename Callable>
void SessionImpl::invoke(Callable &&func)
{
    QMetaObject::invokeMethod(this, std::forward(func));
}
```

Reply:
PS: I mainly wanted to show that we can drop the std::function dependency here. Whether to use std::move or std::forward is another story.

Reply:
I have removed it from this PR, and I will do it in another way where it is needed.